src/pages/api/profile/get-address.ts

```diff
@@ -0,0 +1,115 @@
+import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
+import { getProfileAccessToken } from "@src/utils/get-profile-access-token";
+
+const handler: Handler = async (request, response) => {
+  const webErrors = {
+    invalidProfileToken: ERRORS_DATA.INVALID_PROFILE_TOKEN,
+    invalidScopes: ERRORS_DATA.INVALID_SCOPES,
+    profileDataNotFound: ERRORS_DATA.PROFILE_DATA_NOT_FOUND,
+    unreachable: ERRORS_DATA.UNREACHABLE,
+    invalidQueryString: ERRORS_DATA.INVALID_QUERY_STRING,
+    addressNotFound: ERRORS_DATA.ADDRESS_NOT_FOUND,
+  };
+
+  return getTracer().span("get-address handler", async (span) => {
+    const userCookie = await getServerSideCookies<UserCookie>(request, {
+      cookieName: "user-cookie",
+      isCookieSealed: true,
+      cookieSalt: configVariables.cookieSalt,
+    });
+
+    if (!userCookie) {
+      response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+      response.setHeader("location", "/");
+      response.end();
+      return;
+    }
+
+    const addressId = request.query.addressId;
+
+    if (!addressId || typeof addressId !== "string") {
+      throw webErrorFactory(webErrors.invalidQueryString);
+    }
+
+    const profileAccessToken = await getProfileAccessToken(
+      userCookie.access_token,
+    );
+    span.setDisclosedAttribute(
+      "is Connect.Profile access token available",
+      true,
+    );
+
+    const profileClient = initProfileClient(profileAccessToken);
+
+    const { data: profileAddresses } = await profileClient
+      .getAddresses()
+      .catch((error) => {
+        span.setDisclosedAttribute(
+          "is Connect.Profile addresses fetched",
+          false,
+        );
+
+        if (error.response.status === HttpStatus.UNAUTHORIZED) {
+          throw webErrorFactory(webErrors.invalidProfileToken);
+        }
+
+        if (error.response.status === HttpStatus.FORBIDDEN) {
+          throw webErrorFactory(webErrors.invalidScopes);
+        }
+
+        throw webErrorFactory(webErrors.unreachable);
+      });
+    span.setDisclosedAttribute("is Connect.Profile addresses fetched", true);
+
+    const address = profileAddresses.find(
+      (address) => address.id === addressId,
+    );
+
+    if (!address) {
+      throw webErrorFactory(webErrors.addressNotFound);
+    }
+
+    response.statusCode = HttpStatus.OK;
+    response.json({ address });
+  });
+};
+
+const wrappedHandler = wrapMiddlewares(
+  [
+    tracingMiddleware(getTracer()),
+    rateLimitingMiddleware(getTracer(), logger, {
+      windowMs: 300000,
+      requestsUntilBlock: 200,
+    }),
+    recoveryMiddleware(getTracer()),
+    sentryMiddleware(getTracer()),
+    errorMiddleware(getTracer()),
+    loggingMiddleware(getTracer(), logger),
+    authMiddleware(getTracer()),
+  ],
+  handler,
+  "/api/profile/get-address",
+);
+
+export default new Endpoint<NextApiRequest, NextApiResponse>()
+  .get(wrappedHandler)
+  .getHandler();
```

Comment:
[3/3]
An since we already use the `Endpoint` class from fwl/web to deal with that, it should then be:
```suggestion
export default new Endpoint<NextApiRequest, NextApiResponse>()
  .get(wrappedHandler)
  .patch(/** the future patch handler /*)
  .getHandler();
```

Reply:
(And I think you see me coming for the `identity`'s api pages 😁 )

Reply:
I just saw this comment. Going back to a REST like handling in our file maybe be better 🤔 .

Reply:
Yeah, that's my point, file-wise it would mean a cleaner folder structure (and other things, of course, but it's a part of it)

Reply:
This seems cool, no?

Reply:
Yep, I like it.

Reply:
https://github.com/fewlinesco/connect-account/pull/526#discussion_r623832669 😇

Reply:
Yup, saw that but if you both agree in this thread, I don't see the remaining question in the other. 🤔

Reply:
> we agreed with fenn many months ago to make an API handler for each HTTP verbs, instead of making one which check the verb asked.

I assume then taht it's the change of context @Fuzznimp talked about and we should now use ☝️ ?

Reply:
Yes, the `Endpoint` was designed to answer this question and the fact that reading that inside the `if`s was a bit weird

Reply:
I'll definitely change it then :).

Reply:
👍 🙇

Reply:
I made the changes, and created a task to refacto the rest :).

Reply:
I'll keep these comments unresolved as I point to them in the task.